Astronomy/View Controllers/PhotosCollectionViewController.swift

```diff
@@ -60,19 +60,54 @@ class PhotosCollectionViewController: UIViewController, UICollectionViewDataSour
         return UIEdgeInsets(top: 0, left: 10.0, bottom: 0, right: 10.0)
     }
     
+    func collectionView(_ collectionView: UICollectionView, didEndDisplaying cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        if let cell = cell as? ImageCollectionViewCell {
+            let task = fetchDictionary[cell.photoID]
+            task?.cancel()
+        }
+    }
```

Comment:
Probably a good idea to remove the operation from the dictionary after you cancel it as well.